src/System.Private.CoreLib/shared/System/Decimal.cs

```diff
@@ -292,40 +275,18 @@ public Decimal(int lo, int mid, int hi, bool isNegative, byte scale)
                 flags |= SignMask;
         }
 
-        [OnSerializing]
-        private void OnSerializing(StreamingContext ctx)
-        {
-            // OnSerializing is called before serialization of an object
-            try
-            {
-                // Run the constructor to validate the decimal
-                new decimal(lo, mid, hi, flags);
-            }
-            catch (ArgumentException e)
-            {
-                throw new SerializationException(SR.Overflow_Decimal, e);
-            }
-        }
-
         void IDeserializationCallback.OnDeserialization(object sender)
         {
             // OnDeserialization is called after each instance of this class is deserialized.
             // This callback method performs decimal validation after being deserialized.
-            try
-            {
-                // Run the constructor to validate the decimal
-                new decimal(lo, mid, hi, flags);
-            }
-            catch (ArgumentException e)
-            {
-                throw new SerializationException(SR.Overflow_Decimal, e);
-            }
+            if (!IsValid(flags))
+                throw new SerializationException(SR.Overflow_Decimal);
```

Comment:
This will throw exception without the inner exception now. Does it matter?

Reply:
From an API compatibility perspective it doesn't matter.
Human readability of the exception is somewhat misleading either way (the inner exception was "Decimal byte array constructor requires an array of length four containing valid decimal bytes.").

Reply:
ok